h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostModel.java

```diff
@@ -283,6 +283,15 @@ public XGBoostModel(Key<XGBoostModel> selfKey, XGBoostParameters parms, XGBoostO
     return score0(data, preds, 0.0);
   }
 
+  @Override protected AutoBuffer writeAll_impl(AutoBuffer ab) {
```

Comment:
This will work. There is a bit of duplication though because some of the DataInfo fields are copied to XGBoostOutput. Perhaps the right way is to store DataInfo in XGBoostOutput (GLM does it that way), this would make sure that XGBoostOuput is consistent with DataInfo.

Reply:
That's possible, but would require that {{XGBoostOutputV3}} have custom {{fillImpl}}/{{fillFromImpl}} methods, which in my opinion is more of a maintenance issue.

Reply:
It effectively does have that though...  there is a method 'setDataInfoToOutput(DataInfo dinfo)' that populates those fields, the mapping needs to be somewhere.

I do not think it is a big deal - I am more thinking about the consistent approach across the different algos.